packages/components/src/internal/components/editable/actions.ts

```diff
@@ -116,75 +94,107 @@ export const loadEditorModelData = async (
+    const rows = data.toJS();
+    const orderedRows = dataIds.toArray();
     const forUpdate = loader.mode === EditorMode.Update;
-    if (loader.columns) {
-        columns = editorModel.getColumns(
-            gridData.queryInfo,
-            forUpdate,
-            undefined,
-            loader.columns,
-            loader.columns,
-            colFilter
-        );
+    let columns: QueryColumn[];
+
+    if (loaderColumns) {
+        // TODO: investigate how many loaders actually setting columns, there may be a better path forward
+        columns = loader.columns;
+    } else if (forUpdate) {
+        columns = queryInfo.getUpdateColumns();
     } else {
-        columns = editorModel.getColumns(gridData.queryInfo, forUpdate, undefined, undefined, undefined, colFilter);
+        columns = queryInfo.getInsertColumns();
     }
 
-    const editorModelData = await loadEditorModelData(gridData, columns, forUpdate);
+    // file input columns are not supported in the editable grid, so remove them
+    columns = columns.filter(col => !col.isFileInput);
+
+    // Calculate orderedColumns here before we add PK and Container columns to the columns array because they should
+    // be hidden by default.
+    const orderedColumns = columns.map(queryColumn => queryColumn.fieldKey);
+    const columnMap = columns.reduce((result, column) => {
+        result[column.fieldKey] = column;
+        return result;
+    }, {});
+
+    if (forUpdate) {
+        // If we're updating then we need to ensure that the pkCols and altUpdateKeys are in the columnMap
+        queryInfo.getPkCols().forEach(pkCol => {
+            if (!columnMap[pkCol.fieldKey]) {
+                columnMap[pkCol.fieldKey] = pkCol;
+                columns.push(pkCol);
+            }
+        });
 
-    return {
-        dataModel: dataModel.mutate({
-            ...gridData,
-            rowsLoadingState: LoadingState.LOADED,
-            queryInfoLoadingState: LoadingState.LOADED,
-        }),
-        editorModel: editorModel.merge(editorModelData) as EditorModel,
-    };
-};
+        queryInfo.altUpdateKeys?.forEach(fieldKey => {
+            const col = queryInfo.getColumn(fieldKey);
+            if (col && !columnMap[fieldKey]) {
+                columnMap[fieldKey] = col;
+                columns.push(col);
+            }
+        });
 
-export const initEditableGridModels = async (
-    dataModels: QueryModel[],
-    editorModels: EditorModel[],
-    loaders: EditableGridLoader[],
-    queryModel: QueryModel
-): Promise<EditableGridModels> => {
-    const updatedDataModels = [];
-    const updatedEditorModels = [];
+        const hasContainerCol = columns.filter(c => c.fieldKey === 'Container' || c.fieldKey === 'Folder').length > 0;
 
-    const results = await Promise.all(
-        dataModels.map((dataModel, i) => initEditableGridModel(dataModels[i], editorModels[i], loaders[i], queryModel))
-    );
+        if (!hasContainerCol) {
```

Comment:
why not just always include the containerCol (if it exists in the queryInfo as checked below)? It seems like putting it in the columnMap would be a good idea in all cases instead of relying on the columns to specifically request it.